Source/Editor/Modules/PrefabsModule.cs

```diff
@@ -107,29 +120,35 @@ private void OnPrefabCreated(ContentItem contentItem)
             // Record undo for prefab creating (backend links the target instance with the prefab)
             if (Editor.Undo.Enabled)
             {
-                var selection = Editor.SceneEditing.Selection.Where(x => x is ActorNode).ToList().BuildNodesParents();
-                if (selection.Count == 0)
+                if (!_actor)
                     return;
 
-                if (selection.Count == 1)
+                var actorsList = new List<Actor>();
+                GetActorsTree(actorsList, _actor);
+                
+                var actions = new IUndoAction[actorsList.Count];
+                for (int i = 0; i < actorsList.Count; i++)
                 {
-                    var action = BreakPrefabLinkAction.Linked(((ActorNode)selection[0]).Actor);
-                    Undo.AddAction(action);
-                }
-                else
-                {
-                    var actions = new IUndoAction[selection.Count];
-                    for (int i = 0; i < selection.Count; i++)
-                    {
-                        var action = BreakPrefabLinkAction.Linked(((ActorNode)selection[i]).Actor);
-                        actions[i] = action;
-                    }
-                    Undo.AddAction(new MultiUndoAction(actions));
+                    var action = BreakPrefabLinkAction.Linked(actorsList[i]);
+                    actions[i] = action;
                 }
+                Undo.AddAction(new MultiUndoAction(actions));
+
+                _actor = null;
             }
 
             Editor.Instance.Windows.PropertiesWin.Presenter.BuildLayout();
         }
+        
+        private void GetActorsTree(List<Actor> list, Actor a)
```

Comment:
Could you move this function to `FlaxEditor.Utilities.Utils` as static. Then use it here, in `ActorStaticFlagsEditor` and `ActorLayerEditor` so we reuse more code.